tests/test_cli_subcommand_help_regressions.py

```
@@ -0,0 +1,106 @@
+#!/usr/bin/env python3
+"""Regression test: subcommand --help should never execute command dispatch."""
+
+from __future__ import annotations
+
+import subprocess
+from pathlib import Path
+
+
+def get_repo_root() -> Path:
+    result = subprocess.run(
+        ["git", "rev-parse", "--show-toplevel"],
```

Ruff (0.15.2)

[error] Starting a process with a partial executable path

(S607)